gym/spaces/space.py

```diff
@@ -26,41 +42,41 @@ def __init__(self, shape=None, dtype=None, seed=None):
             self.seed(seed)
 
     @property
-    def np_random(self):
+    def np_random(self) -> np.random.RandomState:
         """Lazily seed the rng since this is expensive and only needed if
         sampling from this space.
         """
         if self._np_random is None:
             self.seed()
 
-        return self._np_random
+        return self._np_random  # type: ignore  ## self.seed() call guarantees right type.
 
     @property
-    def shape(self):
+    def shape(self) -> Optional[Tuple[int, ...]]:
         """Return the shape of the space as an immutable property"""
         return self._shape
 
-    def sample(self):
+    def sample(self) -> T_cov:
         """Randomly sample an element of this space. Can be
         uniform or non-uniform sampling based on boundedness of space."""
         raise NotImplementedError
 
-    def seed(self, seed=None):
+    def seed(self, seed: Optional[int] = None):
```

Comment:
-> list[int]

Reply:
A possible solution to this would be to assert it's an `int` out of all options. But I don't know enough to tell if this will bite us back.